xskillscore/core/deterministic.py

```diff
@@ -291,7 +414,137 @@ def mae(a, b, dim, weights=None, skipna=False):
         b,
         weights,
         input_core_dims=[dim, dim, dim],
-        kwargs={"axis": axis, "skipna": skipna},
-        dask="parallelized",
+        kwargs={'axis': axis, 'skipna': skipna},
+        dask='parallelized',
+        output_dtypes=[float],
+    )
+
+
+def mad(a, b, dim, skipna=False):
+    """
+    Median Absolute Deviation.
+
+    Parameters
+    ----------
+    a : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+    b : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+    dim : str, list
+        The dimension(s) to apply the mae along.
+    skipna : bool
+        If True, skip NaNs when computing function.
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray
+        Median Absolute Deviation.
+
+    See Also
+    --------
+    sklearn.metrics.median_absolute_error
+    xarray.apply_ufunc
+
+    Reference
+    ---------
+    https://en.wikipedia.org/wiki/Median_absolute_deviation
+    """
+    dim, axis = _preprocess_dims(dim)
+
+    return xr.apply_ufunc(
+        _mad,
+        a,
+        b,
+        input_core_dims=[dim, dim],
+        kwargs={'axis': axis, 'skipna': skipna},
+        dask='parallelized',
+        output_dtypes=[float],
+    )
+
+
+def mape(a, b, dim, weights=None, skipna=False):
+    """
+    Mean Absolute Percentage Error.
+
+    Parameters
+    ----------
+    a : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        (Truth which will be divided by)
+    b : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+    dim : str, list
+        The dimension(s) to apply the mae along.
+    weights : xarray.Dataset or xarray.DataArray
+        Weights matching dimensions of ``dim`` to apply during the function.
+        If None, an array of ones will be applied (i.e., no weighting).
+    skipna : bool
+        If True, skip NaNs when computing function.
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray
+        Mean Absolute Percentage Error.
+
+    See Also
+    --------
+    xarray.apply_ufunc
+
```

Comment:
Mind copying the Reference from _mmape e.g.

    Reference
    ---------
    https://en.wikipedia.org/wiki/Mean_absolute_percentage_error